Assets/MixedRealityToolkit.SDK/Experimental/Features/Support/Scripts/BaseServiceManager.cs

```diff
@@ -0,0 +1,395 @@
+        }
+
+        /// <inheritdoc />
+        public IReadOnlyList<T> GetDataProviders<T>(string name = null) where T : IMixedRealityDataProvider
+        {
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                Debug.LogError("This registrar does not support requesting multiple data providers of the same interface type and name.");
+                return new List<T>(); ;
+            }
+
+            Type interfaceType = typeof(T);
+            List<T> matchingProviders = new List<T>();
+
+            IMixedRealityDataProvider[] providers = dataProviders.ToArray();
+            for (int i = 0; i < providers.Length; i++)
+            {
+                if (interfaceType.IsAssignableFrom(providers[i].GetType()))
+                {
+                    matchingProviders.Add((T)providers[i]);
+                }
+            }
+
+            return matchingProviders;
+        }
+
+        /// <inheritdoc />
+        public T GetService<T>(string name = null, bool showLogs = true) where T : IMixedRealityService
+        {
+            T serviceInstance = FindService<T>(name);
+
+            if (showLogs && (serviceInstance == null))
+            {
+                Debug.Log("Failed to get the requested service.");
+            }
+
+            return serviceInstance;
+        }
+
+        /// <inheritdoc />
+        public IReadOnlyList<T> GetServices<T>(string name = null) where T : IMixedRealityService
+        {
+            List<T> services = new List<T>();
+
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                Debug.LogError("This registrar does not support requesting multiple services of the same interface type and name.");
+                return services;
+            }
+
+            T serviceInstance = FindService<T>();
+            if (serviceInstance != null)
+            {
+                services.Add(serviceInstance);
+            }
+            return services;
+        }
+
+        /// <inheritdoc />
+        public bool IsDataProviderRegistered<T>(string name = null) where T : IMixedRealityDataProvider
+        {
+            return (GetDataProvider<T>(name) != null);
+        }
+
+        /// <inheritdoc />
+        public bool IsServiceRegistered<T>(string name = null) where T : IMixedRealityService
+        {
+            return (GetService<T>(name) != null);
+        }
+
+        /// <inheritdoc />
+        public virtual bool RegisterDataProvider<T>(T dataProviderInstance) where T : IMixedRealityDataProvider
+        {
+            if ((dataProviderInstance == null) || (dataProviders.Contains(dataProviderInstance))) { return false; }
+
+            dataProviders.Add(dataProviderInstance);
+
+            return true;
+        }
+
+        /// <inheritdoc />
+        public bool RegisterDataProvider<T>(Type concreteType, SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), params object[] args) where T : IMixedRealityDataProvider
```

Comment:
>-1 [](start = 124, length = 2)

Is this supposed to be "all" or "none"?

Reply:
All. This is the same semantic used elsewhere (ex: MRTK object)